lib/redis-session-store.rb

```diff
@@ -100,107 +68,122 @@ def verify_handlers!
   end
 
   def prefixed(sid)
+    return nil unless sid && sid.private_id
+    "#{default_options[:key_prefix]}#{sid.private_id}"
+  end
+
+  def prefixed_fallback(sid)
     "#{default_options[:key_prefix]}#{sid}"
   end
 
-  def session_default_values
-    [generate_sid, USE_INDIFFERENT_ACCESS ? {}.with_indifferent_access : {}]
+  def create_sid(req)
+    sid = generate_sid
+
+    req.env['redis_session_store.new_session'] = true
+    sid
```

Comment:
would this be the same thing?
```suggestion
    req.env['redis_session_store.new_session'] = true
    generate_sid
```

Reply:
Yep, thank you, too much re-writing 😀